### PR TITLE
CIS WAF rule development

### PR DESCRIPTION
**Pull request by KumarGanesanIBM**

### CODE REVIEW

File: ibm/resource_ibm_cis_waf_rule.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneID = core.StringPtr(zoneID)
+	ruleID := d.Get(cisWAFRuleID).(string)
+	packageID, _, _, _ := convertTfToCisThreeVar(d.Get(cisWAFRulePackageID).(string))
+
+	if d.HasChange(cisWAFRuleMode) {
+		mode := d.Get(cisWAFRuleMode).(string)
+
+		getOpt := cisClient.NewGetWafRuleOptions(packageID, ruleID)
+		getResult, getResponse, err := cisClient.GetWafRule(getOpt)
```
Comment (kavya498) [inline ibm/resource_ibm_cis_waf_rule.go:136]: Mode is provided by user.. why is get performed here?

File: website/docs/r/cis_waf_rule.html.markdown [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+- `cis_id` - (Required,string) The ID of the CIS service instance.
+- `domain_id` - (Required,string) The ID of the domain to change TLS settings.
+- `package_id` - (Required,string) The ID of waf rule package. This field can not be modified.
+- `rule_id` - (Required,string) The ID of waf rule. This field can not be modified.
+- `mode` - (Required,string) The mode setting. This field only once can be set. Valid values: `on`, `off`, `default`, `disable`, `simulate`, `block`, `challenge`.
```
Comment (kavya498) [inline website/docs/r/cis_waf_rule.html.markdown:33]: Add more description like `The mode to use when the rule is triggered. Value is restricted based on the allowed_modes of the rule`

File: examples/ibm-cis/README.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ These types of resources are supported:
 * [ CIS Custom Page](https://cloud.ibm.com/docs/terraform?topic=terraform-cis-resources#cis-custom-page)
 * [ CIS Page Rule](https://cloud.ibm.com/docs/terraform?topic=terraform-cis-resources#cis-page-rule)
 * [ CIS WAF Package ](https://cloud.ibm.com/docs/terraform?topic=terraform-cis-resources#cis-waf-package)
+* [ CIS WAF Rule](https://cloud.ibm.com/docs/terraform?topic=terraform-cis-resources#cis-waf-rule)
```
Comment (kavya498) [inline examples/ibm-cis/README.md:24]: What is the difference between `WAF Package, Waf group` and WAF Rule?

File: ibm/resource_ibm_cis_waf_rule.go (hunk 1 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_waf_rule.go:136]: Need to find owasp or cis mode.  Based on that, we need to set the mode. That's why I read the waf rule to find the mode.

File: website/docs/r/cis_waf_rule.html.markdown (hunk 2 above)
Comment (KumarGanesanIBM) [reply website/docs/r/cis_waf_rule.html.markdown:33]: modified

File: examples/ibm-cis/README.md (hunk 3 above)
Comment (KumarGanesanIBM) [reply examples/ibm-cis/README.md:24]: WAF Rules are in huge amount.  So, WAF Package and WAF Group are for segregate and identify what kind of WAF Rules.

**Pull request merged**